tests/phpunit/tests/functions/__wpCheckAlternateFileNames.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * @group functions.php
+ * @covers ::_wp_check_existing_file_names
+ */
+class Tests_Functions__WpCheckAlternateFileNames extends WP_UnitTestCase {
+
+	/**
+	 * @dataProvider data__wp_check_alternate_file_names
+	 *
+	 * @ticket 55192
+	 *
+	 * @param $filenames Array of filenames to check.
+	 * @param $dir The directory to check.
+	 * @param array $files An array of existing files in the directory.
+	 * @param boolean $expected Expected result.
+	 */
+	public function test__wp_check_alternate_file_names( $filenames, $dir, $files, $expected ) {
+
+		$this->assertSame( $expected, _wp_check_alternate_file_names( $filenames, $dir, $files ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data__wp_check_alternate_file_names() {
+		return array(
+			'no file'          => array(
+				'filename' => array( 'filename.php' ),
```

Comment:
- The array key should be `filenames` instead of `filename`.
- This should be changed in all of the datasets.
- Realignment of array elements will be necessary after making this change in order to meet coding standards.